tests/benchdnn/graph/bench_graph.cpp

```diff
@@ -36,18 +36,27 @@ void check_correctness(const settings_t &s) {
     for (const auto &i_mb : s.mb) {
         deserialized_graph_t dg;
         dg.load(locate_file(s.json_file));
-        flex_rewrite_t fw(i_in_shapes, i_op_attrs, i_fpmath_mode, i_mb, i_dt,
-                i_dt_map, i_op_kind_map);
-        fw.rewrite(dg);
-        BENCHDNN_PRINT(7, "[INFO] Graph dump:\n%s\n", dg.get_string().c_str());
 
-        const prb_t prb(dg, i_expected_n_partition);
+        res_t res {};
         const auto &cpp_pstr = case_to_str(s.json_file, i_in_shapes, i_op_attrs,
                 i_fpmath_mode, i_expected_n_partition, i_mb, i_dt, i_dt_map,
                 i_op_kind_map);
         const char *pstr = cpp_pstr.c_str();
+
+        // rewrite the graph
+        flex_rewrite_t fw(i_in_shapes, i_op_attrs, i_fpmath_mode, i_mb, i_dt,
+                i_dt_map, i_op_kind_map);
+        if (fw.rewrite(dg) != OK) {
+            res.state = UNTESTED;
+            res.reason = "Rewriting unsupported";
+            parse_result(res, pstr);
+
+            continue;
+        }
```

Comment:
Not necessary this suggestion should make into this PR but please consider the following:
```
        if (fw.rewrite(dg) != OK) { // This could be the opposite: dg.rewrite(fw), where `fw` can be `const &` since all the state should be updated in the ctor. This way it gives more clarify which object is modified.
            res.state = UNTESTED;
            res.reason = "Rewriting unsupported";
            parse_result(res, pstr); // All `res` handling can be just replaced with a single benchdnn_print message.

            continue; // Since each case is reset, it should be safe to drop this.
        }
```